### PR TITLE
vimPlugins.helpview-nvim: init at 2024-08-12

### PR DESCRIPTION
**Pull request by nebunebu**
## Description of changes

add vimPlugins.helpview-nvim

## Things done

```sh
vim-plugins-updater -n add "OXY2DEV/helpview.nvim"
vim-plugins-updater -n -p 32 --debug "DEBUG" update
git add -p # staging relevant changes
git commit -m "vimPlugins.helpview-nvim: init at 2024-08-12"
git push --set-upstream origin $(git branch --show-current)
```
If someone has recommendations on a better way to do this, I would appreciate it.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by nebunebu** [comment]
@GaetanLepage

File: pkgs/applications/editors/vim/plugins/generated.nix [hunk 1]
```diff
@@ -4678,6 +4678,19 @@ final: prev:
...
 
+  helpview-nvim = buildVimPlugin {
+    pname = "helpview.nvim";
+    version = "2024-08-12";
+    src = fetchFromGitHub {
+      owner = "OXY2DEV";
+      repo = "helpview.nvim";
+      rev = "49398d628d74283c5a389a6ced40bd8d2daaf56c";
+      sha256 = "0b1z3gz1jpqzf71kjpkw9q3fxbjcdykxi65an6npci0acia49ga2";
+      fetchSubmodules = true;
```
Comment (GaetanLepage) [inline pkgs/applications/editors/vim/plugins/generated.nix:4689]:
```suggestion
```

I am not sure we need to fetch the "wiki" section of the repo.

**Review by GaetanLepage — DISMISSED** [review]
The plugin requires the treesitter parser for vimdoc (https://github.com/OXY2DEV/helpview.nvim/?tab=readme-ov-file#installation). I don't know if we should add `nvim-treesitter` to the dependencies.
@teto do you have an opinion ?

**Comment by teto** [comment]
it's using nvim-treesitter here https://github.com/OXY2DEV/helpview.nvim/blob/c5e6446135a2ef9790f543d21a2b4aff68b6a020/ftplugin/help.lua#L4 but seems like it can fallback just fine. vimdoc is builtins in latest neovim https://github.com/neovim/neovim/blob/master/cmake.deps/deps.txt so it's not necessary.

**Comment by GaetanLepage** [comment]
> it's using nvim-treesitter here https://github.com/OXY2DEV/helpview.nvim/blob/c5e6446135a2ef9790f543d21a2b4aff68b6a020/ftplugin/help.lua#L4 but seems like it can fallback just fine. vimdoc is builtins in latest neovim https://github.com/neovim/neovim/blob/master/cmake.deps/deps.txt so it's not necessary.

Ok thanks !
Let's not have it then.

File: pkgs/applications/editors/vim/plugins/generated.nix (hunk 1 above)
Comment (PerchunPak) [reply pkgs/applications/editors/vim/plugins/generated.nix:4689]: Update script automatically fetches all submodules, if any found. This line will just return on the next update.

**Comment by PerchunPak** [comment]
Result of `nixpkgs-review pr 334208` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>vimPlugins.helpview-nvim</li>
    <li>vimPluginsUpdater</li>
  </ul>
</details>

**Review by PerchunPak — APPROVED** [review]
> but seems like it can fallback just fine

Can confirm, tested on a clean neovim without any other plugins.

LGTM! Thanks!

**Pull request merged**